src/depackers/arcfs.c

```diff
@@ -110,7 +110,7 @@ static unsigned char *read_file_data(FILE *in,
 	unsigned char *data;
 	int siz = hdrp->compressed_size;
 
-	if (!siz || (data = malloc(siz)) == NULL) {
+	if (siz <= 0 || (data = malloc(siz)) == NULL) {
```

Comment:
Considering this field is unsigned in the header tag struct could this just be made into a `size_t`?

Reply:
Nope, unnecessary. If we have to, we can change it to uint32 and check the `value > INT_MAX`
or something.